src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -89,18 +79,15 @@ private void moveNodeToLast(Entry<K, V> entry) {
     public void put(K key, V value) {
         if (data.containsKey(key)) {
             final Entry<K, V> existingEntry = data.get(key);
-            existingEntry.setValue(value);
             moveNodeToLast(existingEntry);
+            existingEntry.setValue(value); 
             return;
         }
-        Entry<K, V> newEntry;
-        if (data.size() == cap) {
+        Entry<K, V> newEntry = new Entry<>(); 
+        if (data.size() >= cap) {
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Potential cache capacity overflow</b></div>
<div id="fix">

The condition `data.size() >= cap` might allow the cache to exceed its capacity by one entry. Consider using `data.size() == cap` to strictly maintain the specified capacity.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -87 +87 @@
 -        if (data.size() >= cap) {
 +        if (data.size() == cap) {
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rishabhbohra3/Algo-tutorial-test/pull/6#issuecomment-2475963128>#730bf5</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged